cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -402,6 +443,45 @@ func NewControllerInitializers(loopMode ControllerLoopMode) map[string]InitFunc
 	controllers["ttl-after-finished"] = startTTLAfterFinishedController
 	controllers["root-ca-cert-publisher"] = startRootCACertPublisher
 
+	// cloud controllers
+	_, withMigration := controllerConfig.ControllerConfig("service-controller")
```

Comment:
I believe we should be checking every controller through this acting appropriately. We have already decided that certain controllers (Eg. node-ipam-controller for GCP) will be running in KCM for most cloud providers but may run in the CCM for certain cloud providers.